client/driver/executor/executor.go

```diff
@@ -0,0 +1,217 @@
+package executor
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+	"sync"
+	"syscall"
+	"time"
+
+	cgroupConfig "github.com/opencontainers/runc/libcontainer/configs"
+
+	"github.com/hashicorp/nomad/client/allocdir"
+	"github.com/hashicorp/nomad/client/driver/env"
+	"github.com/hashicorp/nomad/nomad/structs"
+)
+
+// ExecutorContext is a wrapper to hold context to configure the command user
+// wants to run
+type ExecutorContext struct {
+	TaskEnv          *env.TaskEnvironment
+	AllocDir         *allocdir.AllocDir
+	TaskName         string
+	TaskResources    *structs.Resources
+	FSIsolation      bool
+	ResourceLimits   bool
+	UnprivilegedUser bool
+}
+
+// ExecCommand is a wrapper to hold the user command
```

Comment:
`// ExecCommand is the user command to run. It is a lightweight replacement for exec.Cmd for serialization purposes.`